core/domain/stats_services.py

```diff
@@ -1024,3 +1025,198 @@ def _get_calc_output(exploration_id, state_name, calculation_id):
             calculation_output)
     else:
         return None
+
+
+def change_learner_answer_details(
+        state_reference, entity_type, interaction_id, change):
+    """Changes the learner answer details object, as per the changes received
+    and then returns it.
+
+    Args:
+        state_reference: str. This is used to refer to a state
+            in an exploration or question. Like for an exploration the
+            value will be equal to 'exp_id.state_name' & for question
+            this will be equal to 'question_id' only.
+        entity_type: str. The type of entity i.e 'exploration' or 'question '.
+        interaction_id: str. The ID of the interaction.
+        change: LearnerAnswerDetailsChange. The change to be applied
```

Comment:
Oh, is it a single change? I thought it would be a changelist and this function would iterate through those and apply all of them. Have it like that since that would be more general.

Reply:
Yeah its a single change.